### PR TITLE
article about isolation metrics and complexity

### PR DESCRIPTION
**Pull request by djwillia**

### CODE REVIEW

**Comment by lumjjb** [comment]
Can you rebase master? Tks!

**Comment by lumjjb** [comment]
nvm it's rebased now.

File: _posts/2018-08-03-metrics.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+issues/discussion points you may have there!
+
+
+[^1]: For more information about this way of thinking about container
+      isolation, check out our HotCloud '18 paper [Say Goodbye to
+      Virtualization for a Safer
+      Cloud](https://www.usenix.org/system/files/conference/hotcloud18/hotcloud18-paper-williams.pdf)
+      or James' blog post about the [Horizontal Attack Profile in the
+      cloud](https://blog.hansenpartnership.com/containers-and-cloud-security/).
+
```
Comment (lumjjb) [inline _posts/2018-08-03-metrics.md:82]: Nit: Cut whitespace

File: _posts/2018-08-03-metrics.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+### Counting kernel code
+
+Our first stab at an isolation metric is based on counting the amount
+kernel code needed to support various containers.  We make the
+assumption that the less kernel code we observe is a proxy for less
+complexity, which is a proxy for less bugs/vulnerabilities, which is a
+proxy for greater assurance that the underlying kernel code is
+maintaining isolation.
+
+We use the kernel's `ftrace` mechanism to trace container-related
```
Comment (lumjjb) [inline _posts/2018-08-03-metrics.md:32]: Can we add a link to ftrace https://lwn.net/Articles/365835/ , or https://elinux.org/Ftrace

File: _posts/2018-08-03-metrics.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+of how much nabla containers improve isolation over other mechanisms
+based on this metric:
+
+![graph-functions]({{"public/img/graph-functions.png" | relative_url}})
+
+### Factoring in code complexity
+
+However, an increased number functions in the kernel does not always
+imply more complexity because some functions are more complex than
+others.  As a next step, we used the GNU `complexity` tool to
```
Comment (lumjjb) [inline _posts/2018-08-03-metrics.md:47]: Add link to https://www.gnu.org/software/complexity/?

**Review by lumjjb — CHANGES_REQUESTED** [review]
Add some links + nits

File: _posts/2018-08-03-metrics.md (hunk 1 above)
Comment (djwillia) [reply _posts/2018-08-03-metrics.md:82]: why do you want to cut the whitespace?  Doesn't it make the footnote look cleaner in the markdown?

**Comment by djwillia** [comment]
@lumjjb thank you for the review!  added the links, not sure what you mean about the whitespace

**Review by lumjjb — APPROVED** [review]
LGTM

**Pull request merged**